frontend/src/components/App.jsx

```diff
@@ -43,7 +50,7 @@ class App extends React.Component {
   getSearch(params) {
     return (
       <div>
-        <Header/>
+        <Header onSearchCommit={(commit) => this.setState({commit})}/>
```

Comment:
I am not sure if this is idiomatic React, but the following naming would be more intuitive for me.

`<Header setSearchQuery={(query) => this.setState({query})}/>`

I prefer `query` to `commit` here because I can imagine that in the future querying by other attributes may be supported.